x-pack/legacy/plugins/spaces/server/routes/api/v1/spaces.ts

```diff
@@ -5,16 +5,49 @@
  */
 
 import Boom from 'boom';
+import { Legacy } from 'kibana';
 import { Space } from '../../../../common/model/space';
 import { wrapError } from '../../../lib/errors';
 import { SpacesClient } from '../../../lib/spaces_client';
 import { addSpaceIdToPath } from '../../../lib/spaces_url_parser';
 import { getSpaceById } from '../../lib';
 import { InternalRouteDeps } from '.';
+import { getActiveSpace } from '../../../lib/get_active_space';
+import { getSpaceSelectorUrl } from '../../../lib/get_space_selector_url';
 
 export function initInternalSpacesApi(deps: InternalRouteDeps) {
   const { http, config, spacesService, savedObjects, routePreCheckLicenseFn } = deps;
 
+  http.route({
+    method: 'GET',
+    path: '/api/spaces/v1/activeSpace',
+    async handler(request: Legacy.Request) {
+      const spacesClient: SpacesClient = await spacesService.scopedClient(request);
+      return await getActiveSpace(
+        spacesClient,
+        http.basePath.get(request),
+        config.get('server.basePath')
+      );
+    },
+    options: {
+      pre: [routePreCheckLicenseFn],
+    },
+  });
+
+  http.route({
```

Comment:
@elastic/kibana-platform I know we've previously discussed the demise of `injectedVars` in favor of API calls... have we given any thought to exposing a subset of `kibana.yml` settings to the UI for client-side consumption? This API call is effectively returning the value of `server.basePath`. If we could have that automatically exposed by a core service, then we could eliminate this http request. I assume other plugins could benefit from similar functionality as well.

IIRC, the thought behind removing `injectedVars` was that the NP will have less full-page reloads, so the data we've historically exposed via `injectedVars` could quickly fall outdated, and become useless to the UI. I don't believe settings defined in `kibana.yml` have the same risk though, as these values are meant to be static or very infrequently changed.

Reply:
We have a similar need in https://github.com/elastic/kibana/pull/39477. We need to read the `xpack.security.sessionTimeout` value from the kibana.yml to be able configure HTTP interceptors. It'd be even more awkward here to have to do a HTTP request to setup HTTP interception...

Reply:
I'm not familiar with the `injectedVars` discussion, but on the client-side we expose `coreSetup.http.basePath.get()`.

Reply:
> I'm not familiar with the injectedVars discussion, but on the client-side we expose coreSetup.http.basePath.get().

I believe that's a different base path. The one exposed via the http service is the space-aware base path (i.e., `/serverBasePath/s/marketing`). What I'm specifically interested in is just the `/serverBasePath` part.

Reply:
we are planning to add Client side config in `v7.5`. let's move discussion in the issue https://github.com/elastic/kibana/issues/41990